src/array/cpu/cpu_id_hash_map.h

```diff
@@ -0,0 +1,151 @@
+/**
+ *  Copyright (c) 2019 by Contributors
+ * @file array/cpu/cpu_id_hash_map.h
+ * @brief Class about CPU id hash map
+ */
+
+#ifndef DGL_ARRAY_CPU_CPU_ID_HASH_MAP_H_
+#define DGL_ARRAY_CPU_CPU_ID_HASH_MAP_H_
+
+#include <dgl/aten/types.h>
+
+#include <vector>
+
+#ifdef _MSC_VER
+#include <intrin.h>
+#define COMPARE_AND_SWAP(ptr, oldval, newval, ret)                         \
+  do {                                                        \
+    if (sizeof(newval) == 32) {                               \
+      *ret = _InterlockedCompareExchange(                     \
+        reinterpret_cast<LONG*>(ptr), newval, oldval);        \
+    } else if (sizeof(newval) == 64) {                        \
+      *ret = _InterlockedCompareExchange64(                   \
+        reinterpret_cast<LONGLONG*>(ptr), newval, oldval);    \
+    } else {                                                  \
+      LOG(FATAL) << "ID can only be int32 or int64";          \
+    }                                                         \
+  } while (0)
+#elif __GNUC__
+#if __GNUC__ < 4 || (__GNUC__ == 4 && __GNUC_MINOR__ < 1)
```

Comment:
As @BarclayII suggest, our minimum supported GCC version is GCC 5, so this check is unnecessary.

Reply:
remove for fix